packages/flutter_tools/lib/src/commands/migrate_resolve_conflicts.dart

```diff
@@ -0,0 +1,322 @@
+      }
+      printConflictLine(lines[conflict.endLine!], conflict.endLine!);
+      for (int lineNumber = conflict.endLine! + 1; lineNumber <= (conflict.endLine! + contextLineCount).clamp(0, lines.length - 1); lineNumber++) {
+        printConflictLine(lines[lineNumber], lineNumber, color: TerminalColor.grey);
+      }
+
+      logger.printStatus('\nConflict in $localPath.');
+      // Select action
+      String selection = 's';
+      selection = await terminal.promptForCharInput(
+        <String>['o', 'n', 's'],
+        logger: logger,
+        prompt: 'Accept the (o)riginal lines, (n)ew lines, or (s)kip and resolve the conflict manually?',
+        defaultChoiceIndex: 2,
+      );
+
+      switch(selection) {
+        case 'o': {
+          conflict.chooseOriginal();
+          break;
+        }
+        case 'n': {
+          conflict.chooseNew();
+          break;
+        }
+        case 's': {
+          conflict.skip();
+          break;
+        }
+      }
+    }
+  }
+
+  /// Prints a summary of the changes selected and prompts the developer to commit, abandon, or retry
+  /// the changes.
+  ///
+  /// Returns true if changes were accepted or rejected. Returns false if user indicated to retry.
+  Future<bool> verifyAndCommit(List<Conflict> conflicts, List<String> lines, File file, String localPath) async {
+    int originalCount = 0;
+    int newCount = 0;
+    int skipCount = 0;
+
+    String result = '';
+    int lastPrintedLine = 0;
+    bool hasChanges = false;
+    for (final Conflict conflict in conflicts) {
+      if (!conflict.isValid) {
+        continue;
+      }
+      if (conflict.keepOriginal != null) {
+        hasChanges = true; // don't unecessarily write file if no changes were made.
+      }
+      for (int lineNumber = lastPrintedLine; lineNumber < conflict.startLine!; lineNumber++) {
+        result += '${lines[lineNumber]}\n';
+      }
+      if (conflict.keepOriginal == null) {
+        // Skipped this conflict. Add all lines.
+        for (int lineNumber = conflict.startLine!; lineNumber <= conflict.endLine!; lineNumber++) {
+          result += '${lines[lineNumber]}\n';
+        }
+        skipCount++;
+      } else if (conflict.keepOriginal!) {
+        // Keeping original lines
+        for (int lineNumber = conflict.startLine! + 1; lineNumber < conflict.dividerLine!; lineNumber++) {
+          result += '${lines[lineNumber]}\n';
+        }
+        originalCount++;
+      } else {
+        // Keeping new lines
+        for (int lineNumber = conflict.dividerLine! + 1; lineNumber < conflict.endLine!; lineNumber++) {
+          result += '${lines[lineNumber]}\n';
+        }
+        newCount++;
+      }
+      lastPrintedLine = (conflict.endLine! + 1).clamp(0, lines.length);
+    }
+    for (int lineNumber = lastPrintedLine; lineNumber < lines.length; lineNumber++) {
+      result += '${lines[lineNumber]}\n';
+    }
+    result.trim();
```

Comment:
strings in Dart are immutable; I think what you want here is `result = result.trim()`.

Reply:
Or trimRight()? we probably don't want to trim the start of the string, right?

Reply:
Oops totally missed this. Thanks.

Reply:
Turns out the correct thing to do was to *not* trim. The bug was a feature.